pkg/apis/pipeline/v1/pipeline_types.go

```diff
@@ -352,9 +353,41 @@ func (pt *PipelineTask) GetMatrixCombinationsCount() int {
 	if !pt.IsMatrixed() {
 		return 0
 	}
+	// Explicit combinations in the Matrix
+	// Calculate based on length of include Names and return count
+	if pt.Matrix.MatrixHasInclude() && !pt.Matrix.MatrixHasParams() {
+		return len(pt.Matrix.Include)
+	}
+	// Create a set of param names
+	paramNames := make(map[string][]string)
 	count := 1
+	// Iterate over matrix.params and compute count of all generated combinations
 	for _, param := range pt.Matrix.Params {
 		count *= len(param.Value.ArrayVal)
+		paramNames[param.Name] = param.Value.ArrayVal
+	}
+	if pt.Matrix.MatrixHasInclude() {
+		// Filter out which include params will be replaced
+		for _, include := range pt.Matrix.Include {
+			for _, param := range include.Params {
+				// Check if specific param name is in mapped paramNames
+				if val, exist := paramNames[param.Name]; exist {
+					// If param value exists in mapping, it will overwrite existing param
+					// No new combinations generated
+					if slices.Contains(val, param.Value.StringVal) {
+						break
+					}
+				} else {
+					// Param name not found in mapping
+					// Common param will overwrite all params
+					// No new combinations generated
+					break
+				}
+				// Otherwise the param value does not exist
+				// New combination generated
+				count++
```

Comment:
suggest moving this block upwards:
```
if val, exist := paramNames[param.Name]; exist {
	// If param value exists in mapping, it will overwrite existing param
	// No new combinations generated
	if slices.Contains(val, param.Value.StringVal) {
		break
	} else {
	        // Otherwise the param value does not exist
		// New combination generated
		count++
        }
}
```
I think this is slightly more readable

Reply:
I don't think this works, but I was able to refactor and make it more readable.